python/ray/serve/tests/unit/test_batching.py

```diff
@@ -237,6 +237,26 @@ async def no_op(requests):
             time.sleep(0.25)
 
 
+@pytest.mark.asyncio
+async def test_batch_wait_queue_exceeds_batch_size_race_condition():
```

Comment:
Does this test reliably fail without your fix?

Reply:
Yep